src/number_theory/bernoulli_polynomials.lean

```diff
@@ -0,0 +1,238 @@
+/-
+Copyright (c) 2021 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Ashvni Narayanan
+-/
+
+import number_theory.bernoulli
+
+/-!
+# Bernoulli polynomials
+
+The Bernoulli polynomials are an important tool obtained from Bernoulli numbers.
+
+## Mathematical overview
+
+The $n^{th}$ Bernoulli polynomial is defined as
+$$ B_n(X) = ∑_{k = 0}^n {n χoose k} (-1)^k * B_k * X^{n - k} $$
+where $B_k$ is the $k^{th}$ Bernoulli number. The Bernoulli polynomials are generating functions,
+$$ t * e^{tX} / (e^t - 1) = ∑_{n = 0}^{∞} B_n(X) * ¼{t^n}{n!} $$
+
+## Implementation detail
+
+The Bernoulli (negative) numbers are first defined, `bernoulli_neg`, in order to make proving
+theorems using Bernoulli polynomials easier.
+
+## Main theorems
+
+`sum_bernoulli_neg : ∑ k in range n, (n.choose k : ℚ) * bernoulli_neg k = 0`
+`sum_bernoulli_poly : ∑ k in range (n + 1), ((n + 1).choose k : ℚ) • bernoulli_poly k = `
+  `polynomial.monomial n (n + 1 : ℚ)`
+`exp_bernoulli_poly : power_series.mk (λ n, (polynomial.aeval t ((1 / nat.factorial n : ℚ) • `
+`  bernoulli_poly n))) * (exp A - 1) = X * rescale t (exp A) `
```

Comment:
@bryangingechen: can we ignore the line length linter here, having this format as separate code blocks seems undesirable (assuming that's what happens).

Reply:
Sure, you'd just have to add some lines to [`scripts/style-exceptions.txt`](https://github.com/leanprover-community/mathlib/blob/master/scripts/style-exceptions.txt).

Do we really want to just restate the types in the markdown? The declarations should be just a click away on the web docs if things are properly linked. I think I'd prefer to see short English descriptions of the results there instead.

Reply:
I have changed it, hope this works!

Reply:
Yes, thanks!